doc/development/style_guide.md

```diff
@@ -0,0 +1,870 @@
+For example, use this:
+
+```python
+# %option G_OPT_V_INPUT
+# %end
+# %option G_OPT_R_OUTPUT
+# %end
+```
+
+If needed, override values which need to be different:
+
+```python
+# %option G_OPT_V_INPUT
+# % key: point_input
+# % label: Name of input vector map with points
+# % description: Points used for sampling the raster input
+# %end
+# %option G_OPT_R_OUTPUT
+# % key: raster_input
+# % label: Name of sampled raster map
+# % description: Raster map which will be sampled by the points
+# %end
+```
+
+Do not repeat the values when a standard option defines them.
+
+#### Consider both Flags and Options to modify behavior
+
+Flags are boolean options that default to false. Their names
+are only one character. They are defined using:
+
+```python
+# %flag
+# % key: n
+# % description: Consider zeros to be null values
+# %end
+```
+
+On the command line, the flag is used with dash as `-n`. In Python, the flag
+would be used in the _flags_ parameter of `run_command``:
+
+```python
+gs.run_command(..., flags="n", ...)
+```
+
+However, options are often better because they improve readability, clarify the
+default behavior, and allow for extension of the interface.
+
+**Example:** Consider a tool which by default produces human-readable plain
+text output. Then you add JSON output which is
+enabled by a flag `j`. Later, you decide to add YAML output. This now needs to
+be flag `y` which needs to be exclusive with flag `j`. Soon, you have several
+related flags each exclusive with all the others. Using an option instead of a
+flag from the beginning allows the interface to accommodate more formats. In
+this example, an option named `format` can have default value `plain` and `json`
+for JSON output. When you later add YAML, you simply add `yaml` to the possible
+values without a need for additional options or flags. The interface definition
+for the example would look like:
+
+```python
+# %option
+# % key: format
+# % type: string
+# % required: yes
+# % options: plain,json,yaml
+# % label: Output format
+# % descriptions: plain;Plain text output;json;JSON output;yaml;YAML output
+# % answer: plain
+# %end
+```
+
+#### Adding description and keywords
+
+Each tool needs to have a description and at least 3 keywords:
+
+<!-- markdownlint-disable line-length -->
+
+```python
+# %module
+# % label: Generates a raster map using gaussian random number generator.
+# % description: Mean and standard deviation of gaussian deviates can be expressed by the user.
+# % keyword: raster
+# % keyword: surface
+# % keyword: random
+# %end
+```
+
+<!-- markdownlint-enable line-length -->
+
+Notes:
+
+- the **first** keyword is the module family which goes to the
+  [module family index](https://grass.osgeo.org/grass-stable/manuals/general.html)
+  in the manual
+- the **second** keyword is the overall topic which goes to the
+  [topic index](https://grass.osgeo.org/grass-stable/manuals/topics.html) in the
```

Comment:
Stable or devel for contributing guidelines?

Reply:
Good point